src/uwtools/tests/drivers/test_chgres_cube.py

```diff
@@ -0,0 +1,213 @@
+
+
+def test_ChgresCube_dry_run(config_file, cycle):
+    with patch.object(chgres_cube, "dryrun") as dryrun:
+        driverobj = chgres_cube.ChgresCube(
+            config_file=config_file, cycle=cycle, batch=True, dry_run=True
+        )
+    assert driverobj._dry_run is True
+    dryrun.assert_called_once_with()
+
+
+def test_ChgresCube_namelist_file(driverobj):
+    @external
+    def ready(x):
+        yield x
+        yield asset(x, lambda: True)
+
+    src = str(driverobj._rundir / "input.nml.in")
+    dst = driverobj._rundir / "fort.41"
+    assert not dst.is_file()
+    with patch.object(chgres_cube, "file", new=ready):
+        driverobj._driver_config["namelist"]["update_values"]["config"] = {
+            "data_dir_input_grid": src,
+            "atm_files_input_grid": src,
+            "grib2_file_input_grid": src,
+            "sfc_files_input_grid": src,
+            "mosaic_file_target_grid": src,
+            "varmap_file": src,
+            "vcoord_file_target_grid": src,
+        }
+        driverobj.namelist_file()
+    assert dst.is_file()
+    assert isinstance(f90nml.read(dst), f90nml.Namelist)
+
+
+def test_ChgresCube_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        namelist_file=D,
+        runscript=D,
+    ) as mocks:
+        driverobj.provisioned_run_directory()
+    for m in mocks:
+        mocks[m].assert_called_once_with()
+
+
+def test_ChgresCube_run_batch(driverobj):
+    with patch.object(driverobj, "_run_via_batch_submission") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_ChgresCube_run_local(driverobj):
+    driverobj._batch = False
+    with patch.object(driverobj, "_run_via_local_execution") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_ChgresCube_runscript(driverobj):
+    dst = driverobj._rundir / "runscript.chgres_cube"
+    assert not dst.is_file()
+    driverobj.runscript()
+    with open(dst, "r", encoding="utf-8") as f:
+        lines = f.read().split("\n")
+    # Check directives:
+    assert "#SBATCH --account=me" in lines
+    assert "#SBATCH --time=00:02:00" in lines
+    # Check environment commands:
+    assert "cmd1" in lines
+    assert "cmd2" in lines
+    # Check execution:
+    assert "srun --export=ALL --ntasks $SLURM_CPUS_ON_NODE /path/to/chgres_cube" in lines
+    assert "test $? -eq 0 && touch %s/done" % driverobj._rundir
+
+
+def test_ChgresCube__run_via_batch_submission(driverobj):
+    runscript = driverobj._runscript_path
+    with patch.object(driverobj, "provisioned_run_directory") as prd:
+        with patch.object(
+            chgres_cube.ChgresCube, "_scheduler", new_callable=PropertyMock
+        ) as scheduler:
+            driverobj._run_via_batch_submission()
+            scheduler().submit_job.assert_called_once_with(
+                runscript=runscript, submit_file=Path(f"{runscript}.submit")
+            )
+        prd.assert_called_once_with()
```

Comment:
Now that `_run_via_batch_submission()` and `_run_via_local_execution()` (below) are part of the `Driver` base class, could we consider having just one copy of the test methods, in `uwtools.tests.drivers.test_driver`, and remove the (I think) duplicates in each of the `Driver` subclass test modules? Since the `ConcreteDriver` class defined in `test_driver.py` inherits those `_run_*()` methods, it seems like it should be possible to test them there, once and for all `Driver`s. Then we could potentially remove any duplicates.